auth/client/src/main/java/org/wildfly/security/auth/client/spi/ArtemisClientConfigProvider.java

```diff
@@ -0,0 +1,140 @@
+import java.util.concurrent.ConcurrentMap;
+import javax.net.ssl.SSLContext;
+import org.apache.activemq.artemis.core.remoting.impl.netty.TransportConstants;
+import org.apache.activemq.artemis.core.remoting.impl.ssl.DefaultSSLContextFactory;
+import org.apache.activemq.artemis.spi.core.remoting.ssl.SSLContextConfig;
+import org.apache.activemq.artemis.spi.core.remoting.ssl.SSLContextFactory;
+import org.apache.activemq.artemis.utils.ConfigurationHelper;
+import org.kohsuke.MetaInfServices;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.AuthenticationContextConfigurationClient;
+import org.wildfly.security.auth.client.ElytronXmlParser;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2021 Red Hat, Inc.
+ */
+@MetaInfServices(value = SSLContextFactory.class)
+public class ArtemisClientConfigProvider extends DefaultSSLContextFactory {
+
+    static final AuthenticationContextConfigurationClient AUTH_CONTEXT_CLIENT = AccessController.doPrivileged((PrivilegedAction<AuthenticationContextConfigurationClient>) AuthenticationContextConfigurationClient::new);
+    private static final ConcurrentMap<Object, SSLContext> sslContextCache = new ConcurrentHashMap<>(2);
+
+    @Override
+    public void clearSSLContexts() {
+        sslContextCache.clear();
+    }
+
+    @Override
+    public SSLContext getSSLContext(SSLContextConfig config, Map<String, Object> configuration) throws Exception {
+        final Object cacheKey = getCacheKey(config, configuration);
+        return sslContextCache.computeIfAbsent(cacheKey, key -> {
+            try {
+                SSLContext context = getElytronSSLContext(config, configuration);
+                if (context == null) {
+                    context = super.getSSLContext(config, configuration);
+                }
+                return context;
+            } catch (final Exception ex) {
+                throw new RuntimeException("An unexpected exception occured while creating JDK SSLContext with " + config, ex);
+            }
+        });
+    }
+
+    private SSLContext getElytronSSLContext(SSLContextConfig config, Map<String, Object> configuration) throws Exception {
+        String host = ConfigurationHelper.getStringProperty(TransportConstants.HOST_PROP_NAME, TransportConstants.DEFAULT_HOST, configuration);
+        int port = ConfigurationHelper.getIntProperty(TransportConstants.PORT_PROP_NAME, TransportConstants.DEFAULT_PORT, configuration);
+        String sslContextName = ConfigurationHelper.getStringProperty(TransportConstants.SSL_CONTEXT_PROP_NAME, null, configuration);
+        String path;
+        if (sslContextName != null && !sslContextName.isEmpty()) {
+            path = '/' + sslContextName;
+        } else {
+            path = "";
+        }
+        URI uri = new URI("artemis://" + host + ":" + port + path);
+        String xmlPath = ConfigurationHelper.getStringProperty("wildfly-config-url", null, configuration);
```

Comment:
@ehsavoie I have seen the https://issues.apache.org/jira/browse/ARTEMIS-3756 mentioned in the analysis, but am not sure why additional system property has to be introduced and parsed from the configuration. Are there some integration tests in other PRs for this feature that I can check out to understand better?

Reply:
`This may be an issue for any SPI on the client side since it doesn't get the configuration defined on the server side.` Should the client need server's configuration?

Reply:
@ehsavoie As Diana mentioned, I think some example config (or an example test case) could help us better understand what's needed.